proposals/0016-simd-application-fees.md

```diff
@@ -0,0 +1,438 @@
+account is taken by a transaction batch no other batch can use this account in
+parallel, so only transactions belonging to a single batch are processed
+correctly and all others are retried again or forwarded to the leader of the
+next slot. With all the forwarding and retrying the validator is choked by the
+transactions write-locking the same accounts and effectively processing valid
+transactions sequentially.
+
+There are multiple accounts of OpenBook (formerly serum), mango markets, etc
+which are used in high-frequency trading. During the event of extreme
+congestion, we observe that specialized trading programs write lock these
+accounts but never CPI into the respective programs unless they can extract
+profit, effectively starving the actual users for access. With current low
+cluster fees, it incentivizes spammers to spam the network with transactions.
+
+Without any proper rebate mechanism for users, Solana gas fees will increase and
+it will lose the edge for being low fees cluster. For entities like market
+makers, Solana must remain a low-fee cluster as they have a thin profit margin
+and have to quote quite often. The goal of this proposal is to reduce spamming
+without really increasing fees for everyone. Encourage users to create proper
+transactions. Read the cluster state before creating transactions instead of
+spamming. We are motivated to improve the user experience, and user activity and
+keep Solana a low gas fee blockchain. Keeping gas fees low and increasing user
+activity on the chain will help all the communities based on Solana grow. This
+will also help users to gain more control over accounts owned by them.
+
+Note that this proposal does not aim to increase gas fees for all transactions
+or add any kind of congestion control mechanism.
+
+## Alternatives Considered
+
+* Having a fixed write lock fee and a read lock fee.
+
+Pros: Simpler to implement, Simple to calculate fees.
+
+Cons: Will increase fees for everyone, dapp cannot decide to rebate fees, (or
+not all existing apps have to develop a rebate system). Fees cannot be decided
+by the dapp developer. Penalizes everyone for a few bad actors.
+
+* Passing application fees for each account by instruction and validating them
+  during the execution inside user application code
+
+Pros: High efficiency, minimal performance impact, more generic, no need to
+change the account structure.
+
+Cons: Cannot prevent denial of service attacks, or account blocking
+attacks that do not call into the respective program
+
+## New Terminology
+
+Application Fees: Fees that are decided by the dapp developer will be charged if
+a user wants to use the dapp. They will be applied even if the transaction fails
+contrary to lamport transfers. The program can decide to rebate these fees back
+to the user if certain conditions decided by dapp developers are met.
+
+### Other terminology
+
+`Owner` of an account: It is the account owner as specified in the `owner`
```

Comment:
No need to attempt to redefine an existing term: https://docs.solana.com/terminology#account-owner